### DISCUSSION TITLE
Keeping Settings during Docker Container Replacement

### DISCUSSION THREAD

**Discussion by paswai** [seriousm4x/UpSnap#73] [Help wanted] [answered]
Hi all,
first of all, I really like UpSnap and it is a really helpful tool if one wants devices to run only when needed. I am not that experienced with docker and started to update containers including UpSnap. Naturally, all settings (users, passwords) and devices are gone. Is there a way to keep the settings and devices or define them using env variables?
Thanks for the help!

**Comment by seriousm4x** [comment, marked as answer]
Hi, thats what docker volumes are for. It allows you to mount a folder from your host inside the container.
Use this and your database will be persistant: https://github.com/seriousm4x/UpSnap/blob/master/docker-compose.yml#L8-L9

**Reply by paswai** [reply]
Where is the facepalm emoji, when one needs it? I didn't add the volume. Thank you for the help! 
Have a great Sunday!
